### PR TITLE
cluster: move more functions to cluster builder

### PR DESCRIPTION
**Pull request by ramaraochavali**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -922,11 +851,7 @@ func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings,
 	}
 }
 
-func applyLocalityLBSetting(
-	locality *core.Locality,
-	cluster *cluster.Cluster,
-	localityLB *networking.LocalityLoadBalancerSetting,
-) {
+func applyLocalityLBSetting(locality *core.Locality, cluster *cluster.Cluster, localityLB *networking.LocalityLoadBalancerSetting) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:854]: Weird, shouldn't this return a lint error(lll )

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:854]: sorry did not get. Which lint error?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:854]: I see. Not sure why there are lines longer than this for example `applyUpstreamTLSSettings`

**Pull request merged**